src/apps/frontend/components/header/index.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { useEffect } from 'react';
+import toast from 'react-hot-toast';
+import { useNavigate } from 'react-router-dom';
+
+import { useAccountContext, useAuthContext } from '../../contexts';
+
+import HamburgerToggleButton from './hamburger-toggle-button';
+import UserProfileSnippet from './user-profile-snippet.component';
+
+type HeaderProps = {
+  isSidebarOpen: boolean;
+  setIsSidebarOpen: (open: boolean) => void;
+};
+
+const Header: React.FC<HeaderProps> = ({
+  isSidebarOpen,
+  setIsSidebarOpen,
+}) => {
+  const navigate = useNavigate();
+  const { logout } = useAuthContext();
+  const { accountDetails, getAccountDetails } = useAccountContext();
+
+  const handleSidebarToggle = (state: boolean) => {
+    setIsSidebarOpen(state);
+  };
+
+  const handleSignOut = () => {
+    logout();
+    navigate('/login');
+  };
+
+  useEffect(() => {
+    (async () => {
+      await getAccountDetails();
```

Comment:
We want to avoid component making the call and maintain state like this, have you see usage of `useAsync` where context maintains loading, loaded, error etc state for an operation? This helps to make sure data operation state is maintained in the context itself.

Reply:
So sir, should I invoke the get method in the context itself. As I have seen in the `useAsync` hook, it does provides states like `isLoading`, `error` and `result` but for the method it is just providing the `asyncCallback` and we do need to invoke this method somewhere to make the api call and get required data. If we want to avoid get methods being called in a component then either we need to invoke them in either context(using useEffect) or in the `useAsync` hook. What will be the preferred?

Reply:
Its fine to invoke from component, but I want to make sure we are using the state from the context - like isloading, iserror etc instead of catch / then to render the UI diferently as the operation progress / completes

Reply:
okay sir, I will refractor accordingly